portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,59 +28,81 @@
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    /*Query query = new Query("Comments").addSort("time", SortDirection.DESCENDING);
+    Query query = new Query("Comments").addSort("time", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<Task> comments = new ArrayList<>();
+    List<Comment> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
+      if (commentCount == 0) {break;}
       long id = entity.getKey().getId();
-      String title = (String) entity.getProperty("comment");
-      long timestamp = (long) entity.getProperty("time");
+      String Statement = (String) entity.getProperty("comment");
+      long Time = (long) entity.getProperty("time");
 
-      Task message = new Task(id, title, timestamp);
-      comments.add(message);gson.toJson(comments)
+      Comment messages = new Comment(id, Statement, Time);
+      comments.add(messages);
+      commentCount--;
     }
 
-    Gson gson = new Gson();*/
+    Gson gson = new Gson();
 
-    response.setContentType("text/html;");
-    response.getWriter().println(comment);
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
   }
   
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String statement = comment(request, "texts");
       long time = System.currentTimeMillis();
+      String statement = comment(request, "texts");
+      commentCount = maxComments(request, "preferedNumber");
+
+      if (commentCount < 0) {
+        response.setContentType("text/html");
+        response.getWriter().println("Please enter an integer between 0 and 100.");
+        return;
+      }
 
       addcomment(statement, time);
 
       response.sendRedirect("/comments.html");
   }
 
   private String comment(HttpServletRequest request, String message) {
-    String client_comment = request.getParameter(message);
-    return client_comment;
+    return request.getParameter(message);
   }
 
   private void addcomment(String statements, long timestamp) {
 
-    Entity COMMENTS = new Entity("Comments");
-    COMMENTS.setProperty("comment", statements);
-    COMMENTS.setProperty("time", timestamp);
+    Entity comments = new Entity("Comments");
+    comments.setProperty("comment", statements);
+    comments.setProperty("time", timestamp);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(COMMENTS);
+    datastore.put(comments);
   }
 
+  private int maxComments(HttpServletRequest request, String preferredNumber) {
+      String choice = request.getParameter(preferredNumber);
+      int maxNumber;
+      try {
+      maxNumber = Integer.parseInt(choice);
+    } catch (NumberFormatException e) {
```

Comment:
I would expect these to be indented for style (ie the `}` on the same level as the `try {` and the closing `}` for the catch. This won't break anything but it help in readability.